external_parser/example_joiner.cc

```diff
@@ -315,52 +358,73 @@ int example_joiner::process_interaction(const v2::Event &event,
     data.probabilityOfDrop = 1.f - metadata.pass_probability();
     data.skipLearn = cb->deferred_action();
 
-    const char* context = reinterpret_cast<char const *>(cb->context()->data());
-    joined_event je = {
-      enqueued_time_utc,
-      std::move(meta),
-      std::move(data)
-    };
-
-    if (_binary_to_json) {
-      je.context = context;
-      je.model_id = cb->model_id()->data();
-    }
-
+    const char *context = reinterpret_cast<char const *>(cb->context()->data());
     std::string line_vec(context, cb->context()->size());
 
-    if (_vw->audit || _vw->hash_inv) {
-      VW::template read_line_json<true>(
-          *_vw, examples, const_cast<char *>(line_vec.c_str()),
-          reinterpret_cast<VW::example_factory_t>(&VW::get_unused_example), _vw,
-          &_dedup_cache.dedup_examples);
-    } else {
-      VW::template read_line_json<false>(
-          *_vw, examples, const_cast<char *>(line_vec.c_str()),
-          reinterpret_cast<VW::example_factory_t>(&VW::get_unused_example), _vw,
-          &_dedup_cache.dedup_examples);
+    joined_event je{enqueued_time_utc,
+                    {timestamp_to_chrono(*metadata.client_time_utc()),
+                     metadata.app_id() ? metadata.app_id()->str() : "",
+                     metadata.payload_type(), metadata.pass_probability(),
+                     metadata.encoding(), metadata.id()->str(), learning_mode},
```

Comment:
did we check if `metadata.id()` was valid before here?

Reply:
yes it's been checked on a higher level